go-controller/pkg/node/gateway_init.go

```diff
@@ -563,3 +563,26 @@ func (nc *DefaultNodeNetworkController) updateGatewayMAC(link netlink.Link) erro
 	return nil
 
 }
+
+func (nc *DefaultNodeNetworkController) getGatewayBridgeName(nodeAddr net.IP) (string, error) {
+	if config.OvnKubeNode.Mode == types.NodeModeDPUHost {
+		gwIntf, err := getInterfaceByIP(nodeAddr)
+		if err != nil {
+			return "", err
+		}
+		config.Gateway.Interface = gwIntf
```

Comment:
Why does a get function modify the global config?

Reply:
Thanks for the link, I still don't understand why is that needed in `getGatewayBridgeName`

Reply:
getGatewayNextHops function uses config.Gateway.Interface. I think we can expect gateway interface name to be different in case of dpu-host. 
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/node/gateway_init.go#L155-L161 

I agree that I do not have a concrete answer to your question. @trozet might know.